crates/chia_py_streamable_macro/src/lib.rs

```diff
@@ -159,10 +159,23 @@ pub fn py_streamable_macro(input: proc_macro::TokenStream) -> proc_macro::TokenS
     py_protocol.extend(quote! {
         #[pyo3::pymethods]
         impl #ident {
-            #[staticmethod]
+            #[classmethod]
             #[pyo3(signature=(json_dict))]
-            pub fn from_json_dict(json_dict: &pyo3::Bound<pyo3::PyAny>) -> pyo3::PyResult<Self> {
-                <Self as #crate_name::from_json_dict::FromJsonDict>::from_json_dict(json_dict)
+            pub fn from_json_dict(cls: &pyo3::Bound<'_, pyo3::types::PyType>, json_dict: &pyo3::Bound<pyo3::PyAny>) -> pyo3::PyResult<pyo3::PyObject> {
+                use pyo3::prelude::PyAnyMethods;
+                use pyo3::IntoPy;
+                let rust_obj = <Self as #crate_name::from_json_dict::FromJsonDict>::from_json_dict(json_dict);
```

Comment:
I would expect that a `?` at the end of this statement would work. Then you don't need to `match` the `rust_obj` to handle the `Err()` case

Reply:
Done